crates/red_knot_python_semantic/src/types.rs

```diff
@@ -498,21 +502,7 @@ pub(crate) use todo_type;
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, salsa::Update)]
 pub enum Type<'db> {
     /// The dynamic type: a statically unknown set of values
-    Any,
-    /// Unknown type (either no annotation, or some kind of type error).
-    /// Equivalent to Any, or possibly to object in strict mode
-    Unknown,
-    /// Temporary type for symbols that can't be inferred yet because of missing implementations.
-    /// Behaves equivalently to `Any`.
-    ///
-    /// This variant should eventually be removed once red-knot is spec-compliant.
-    ///
-    /// General rule: `Todo` should only propagate when the presence of the input `Todo` caused the
-    /// output to be unknown. An output should only be `Todo` if fixing all `Todo` inputs to be not
-    /// `Todo` would change the output type.
-    ///
-    /// This variant should be created with the `todo_type!` macro.
-    Todo(TodoType),
+    Any(AnyType),
```

Comment:
I'm on my phone so I don't have have an easy way to verify it myself. Does this increase my the overall size of Type in release builds or does it remain unchanged?

Reply:
The static assertion is unmodified:

https://github.com/astral-sh/ruff/blob/b0905c4b043189879af78afb757270c4e0397e25/crates/red_knot_python_semantic/src/types.rs#L4001-L4004

Reply:
The debug size is also the same (32 byte). Which was surprising to me because the `Todo` variant was the largest a while ago (24 byte). But now it has been replaced by `Type::SubclassOf`, which contains `SubclassOfType` of size 24 byte because it can also contain a heavy `Todo`.